src/shared/auth/user.guard.ts

```diff
@@ -0,0 +1,19 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { UserDataStatus } from 'src/subdomains/generic/user/models/user-data/user-data.entity';
+import { UserStatus } from 'src/subdomains/generic/user/models/user/user.entity';
+
+@Injectable()
+export class UserGuard implements CanActivate {
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const req = context.switchToHttp().getRequest();
+    const userStatus = req.user.userStatus;
+    const userDataStatus = req.user.userDataStatus;
+
+    return (
+      userStatus !== UserStatus.BLOCKED &&
+      userStatus !== UserStatus.DELETED &&
+      userDataStatus !== UserDataStatus.BLOCKED &&
+      userDataStatus !== UserDataStatus.DEACTIVATED
+    );
```

Comment:
We should activate this in a second step. Because all existing JWTs will not have the new status fields.

Reply:
Why? That doesn't matter I would say 🤔 if userStatus or userDataStatus is undefined it's valid 👍 it's only invalid/forbidden if the status is blocked, so everything fine I would say, or?

Reply:
True 👍  could do `[UserStatus.BLOCKED, UserStatus.DELETED].includes(userStatus) && ...`